scalalib/src/mill/scalalib/SemanticDbJavaModule.scala

```diff
@@ -198,7 +198,8 @@ trait SemanticDbJavaModule extends CoursierModule { hostModule: JavaModule =>
               scalacOptions = scalacOptions,
               compilerClasspath = m.scalaCompilerClasspath(),
               scalacPluginClasspath = semanticDbPluginClasspath(),
-              reporter = T.reporter.apply(hashCode)
+              reporter = T.reporter.apply(hashCode),
+              reportOldProblems = false
```

Comment:
My initial though was, that semanticdb data is used via Metals, and I think BSP clients are persisting problems anyway. Problems are send as events and not as a result of a compilation, so the client need to manage them anyway.

But OTOH, when semanticdb data is needed for other use, e.g. scalafix, warnings might be still of interest. WDYT?

Reply:
I don't know if Metals can correctly handle duplicated problems.
Maybe @tgodzik can answer this?
About Scalafix, having the warnings reported again would be good.
But this decision can also be taken in a second moment. We can merge this as `false` and then experiment in a second PR to see the implications for BSP.

Reply:
We will show all the diagnostics that we are sent until we get `reset` in PublishDiagnosticsParams , since we don't know if problems were solved until we are told so explicitely.

Reply:
@tgodzik Thanks Tomasz! That means, in case of incremental compilation, diagnostics where until now either incomplete (if there was a `reset`) or may even contain outdated or duplicate problems (if there was no `reset`). I think, always respecting the setting of `zincReportOldProblems` is probably the cleanest option.